canmatrix/canmatrix.py

```diff
@@ -536,60 +534,27 @@ class Frame(object):
     comment
     """
 
-    def __init__(self, name, **kwargs):
-        self.name = name
-            
-        args = [
-            ('Id', '_Id', int, 0),
-            ('dlc', 'size', int, 0),
-            ('transmitter', 'transmitter', None, []),
-            ('extended', 'extended', bool, False),
-            ('is_complex_multiplexed', 'is_complex_multiplexed', bool, False),
-            ('is_fd', 'is_fd', bool, False),
-            ('comment', 'comment', str, None),
-            ('signals', 'signals', None, []),
-            ('mux_names', 'mux_names', None, {}),
-            ('attributes', 'attributes', None, {}),
-            ('receiver', 'receiver', None, []),
-            ('signalGroups', 'signalGroups', None, []),
-            ('j1939_pgn', '_j1939_pgn', int, None),
-            ('j1939_source', '_j1939_source', int, 0),
-            ('j1939_prio', '_j1939_prio', int, 0),
-            ('is_j1939', 'is_j1939', bool, False),
-            #            ('cycleTime', '_cycleTime', int, None),
-            #            ('sendType', '_sendType', str, None),
-        ]
-
-        for arg_name, destination, function, default in args:
-            try:
-                value = kwargs[arg_name]
-            except KeyError:
-                value = default
-            else:
-                kwargs.pop(arg_name)
-            if function is not None and value is not None:
-                value = function(value)
-            setattr(self, destination, value)
-            
-        if len(kwargs) > 0:
-            raise TypeError('{}() got unexpected argument{} {}'.format(
-                self.__class__.__name__,
-                's' if len(kwargs) > 1 else '',
-                ', '.join(kwargs.keys())
-            ))
-
-        if self._j1939_pgn is not None:
-            self.recalcJ1939Id()
-        # we got a j1939 PGN,
 
+    name = attr.ib(default="")
+    Id = attr.ib(default = 0)
+    size = attr.ib(default = 0)
+    transmitter = attr.ib(default =[])
```

Comment:
Don't use mutables as defaults, just like you shouldn't do it as regular function defaults.

https://repl.it/@altendky/why-not-mutable-parameter-defaults

```python3
transmitter = attr.ib(factory=list)
```
This will call `list` each time a new instance is created such that each one gets a unique list.

Side note, this used to be written as `attr.ib(default=attr.Factory(list))`.  Just FYI in case you see it around.

Be sure to look below at the other `[]` and `{}` defaults.